src/pages/DistributePage/DistributePage.tsx

```diff
@@ -101,8 +127,34 @@ function DistributePage() {
       );
 
       if (trx) {
+        const claims: ValueTypes['claims_insert_input'][] = Object.entries(
+          distribution.claims
+        ).map(([address, claim]) => ({
+          address,
+          index: claim.index,
+          amount: Number(
+            FixedNumber.from(BigNumber.from(claim.amount)).divUnsafe(
+              FixedNumber.from(denominator)
+            )
+          ),
+          proof: claim.proof.toString(),
+          user_id: users.find(({ address }) => address === address)?.id,
+        }));
+
+        const updateDistribution: ValueTypes['distributions_insert_input'] = {
+          total_amount: Number(
+            FixedNumber.from(totalDistributionAmount, 'fixed128x18')
+          ),
+          epoch_id: Number(epochId),
+          merkle_root: distribution.merkleRoot,
+          claims: {
+            data: claims,
+          },
+          vault_id: Number(selectedVault.id),
+        };
+        setDistributionDTO(updateDistribution);
+        await mutateAsync();
```

Comment:
i feel nervous about this gap in time between the transaction being submitted & the distribution info being saved to the db... i kind of want to suggest that we save the info to the db first, marked as pending, then submit the tx, & mark as committed afterward. so that the worst case scenario is orphaned data in the db, not an orphaned merkle tree

Reply:
I was uneasy about this as well. I implemented your suggestion and I think I'll sleep better tonight. Gonna create the migration and the update mutation.

Reply:
Added a boolean column to the distributions tables called `saved_on_chain` which is updated after the distribution is saved on chain. Updated table permissions to allow user updates to `distributions` and `claims` table.